File: reverse-string/reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        int left = 0;
        int right = s.length -1;
        while(left < right){
            char first = s[left];
            char second = s[right];
            s[left] = second;
            s[right] = first;
            left++;
            right--;
        }
    }
}

File: reverse-string/reverse-string.kt
class Solution {
    fun reverseString(s: CharArray): Unit {
        var left = 0
        var right = s.size - 1
        
        while(left < right){
            val leftVal = s[left]
            val rightVal = s[right]
            s[left++] = rightVal
            s[right--] = leftVal
        }
    }
}